BEIMA.Backend.Test/MongoService/BaseDeviceTest.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using MongoDB.Bson;
+using NUnit.Framework;
+
+namespace BEIMA.Backend.MongoService.Test
+{
+    [TestFixture]
+    public class BaseDeviceTest
+    {
+        //Instance variables
+        readonly ObjectId validObjId = ObjectId.GenerateNewId();
+        readonly ObjectId validDeviceTypeId = ObjectId.GenerateNewId();
+        readonly string validDeviceTag = "DeviceTag123";
+        readonly string validManufacturer = "BSU";
+        readonly string validModelNum = "ABC-123";
+        readonly string validSerialNum = "ABCDEFG123456";
+        readonly int validYearManufactured = 2022;
+        readonly string validNotes = "Example of notes.";
+
+        readonly DateTime validDate = DateTime.Now.ToUniversalTime();
+        readonly string validUser = "User1";
+
+        readonly ObjectId validBuildingId = ObjectId.GenerateNewId();
+        readonly string validLocationNotes = "Example of location notes.";
+        readonly string validLatitude = "-000.21.21";
+        readonly string validLongitude = "-323.32.32";
+
+        readonly string validKey1 = "key1";
+        readonly string validKey2 = "key2";
+        readonly string validKey3 = "key3";
+        readonly string validValue1 = "value1";
+        readonly bool validValue2 = true;
+        readonly int validValue3 = 123;
+
+        [Test]
+        public void BaseDeviceNotInstantiated_InstantiateUsingFullConstructor_BaseDeviceInstantiatedWithCorrectValues()
+        {
+            var device = new BaseDevice(validObjId, validDeviceTypeId, validDeviceTag, validManufacturer, validModelNum, validSerialNum, validYearManufactured, validNotes);
+            Assert.That(device.Id, Is.EqualTo(validObjId));
+            Assert.That(device.DeviceTypeId, Is.EqualTo(validDeviceTypeId));
+            Assert.That(device.DeviceTag, Is.EqualTo(validDeviceTag));
+            Assert.That(device.Manufacturer, Is.EqualTo(validManufacturer));
+            Assert.That(device.ModelNum, Is.EqualTo(validModelNum));
+            Assert.That(device.SerialNum, Is.EqualTo(validSerialNum));
+            Assert.That(device.YearManufactured, Is.EqualTo(validYearManufactured));
+            Assert.That(device.Notes, Is.EqualTo(validNotes));
+
+            device.SetLastModified(validDate, validUser);
```

Comment:
I think it's good practice to set all of the object stuff at the beginning, and than do all the assertions at the end.

Reply:
Would we want to just pull the assertions at the top out into their own test case?

Reply:
We shouldn't need to, setting the various properties should not be affecting one another.

Reply:
All the fields need to be set for a Bson to be returned correctly. There's another test case that only sets a few fields and leaves the rest null. Moving around the asserts is a good idea.

Reply:
Resolved [bbc776a](https://github.com/cs481-ekh/s22-beima/pull/114/commits/bbc776a0b050b6457993e35109a7704be0e6b827)